spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -10,10 +18,34 @@
     end
 
     context "when sending a welcome_thread notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who has commented in a welcome thread"
+      let(:welcome_thread_article) { create(:article, title: "Welcome Thread - v3000", published: true) }
+      let!(:welcome_thread_comment) { create(:comment, commentable_id: welcome_thread_article.id, commentable_type: "Article", user_id: user.id) }
+
+      it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
+        expect(receiving_user.notifications.count).to eq(0)
+        sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+
+        expect(receiving_user.notifications.count).to eq(1)
+        expect(receiving_user.notifications.first.notifiable).to eq(welcome_broadcast)
+      end
+
+      it "does not send a notification to a user who has commented in a welcome thread", :aggregate_failures do
+        expect(welcome_thread_article.comments.count).to eq(1)
+        expect(welcome_thread_comment.commentable).to eq(welcome_thread_article)
+        expect(user.comments.count).to eq(1)
+
+        sidekiq_perform_enqueued_jobs { described_class.call(user.id) }
+
+        expect(user.notifications.count).to eq(0)
+      end
+
+      it "does not send a duplicate notification" do
+        2.times do
+          sidekiq_perform_enqueued_jobs { described_class.call(receiving_user.id) }
+        end
```

Comment:
This is more of a nitpick but feel free to ignore it :D 

I think you can replace this with:

```ruby
sidekiq_perform_enqueued_jobs do
  2.times { described_class.call(receiving_user.id) }
end
```

basically run the function twice inside the "perform" block, so that the perform only runs once

Reply:
I don't mind nitpicky! I've made this change. 😄